### PR TITLE
Upgrade to Storybook v7

### PR DESCRIPTION
**Pull request by connor-baer**
## Purpose

Storybook v7 (currently in [beta](https://storybook.js.org/blog/7-0-beta/)) is scheduled to be released in late March. Initially, I wanted to wait for the stable release; however, the proof of concept for Linaria (#1620) requires Babel 7 to work, so I decided to upgrade now. There should be no more breaking changes between now and the general release.

## Approach and changes

I followed the [unofficial](https://chromatic-ui.notion.site/Storybook-7-migration-guide-dbf41fa347304eb2a5e9c69b34503937) and [official](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-65x-to-700) migration guides and applied most of the automatic codemods with `npx storybook@next upgrade --prerelease`.

- Upgraded to Babel 7 (a prerequisite for #1620) and simplified the Babel configuration
- Adapted the [new naming convention](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#mdx-docs-files) for docs. MDX files now import stories, not the other way around. Docs and stories are now listed separately in the sidebar.
- Removed the MDX component customization and added global CSS instead
- Added a decorator to apply the system color scheme
- Injected the design system CSS variables globally

### Optional

- [ ] Migrate to CSF 3 (probably better as a follow-up PR)

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [ ] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: .storybook/components/MediaQueriesTable.tsx [hunk 1]
```diff
@@ -13,30 +13,27 @@
...
+import { useTheme, ThemeProvider } from '@emotion/react';
 import { light } from '@sumup/design-tokens';
 import { Table } from '@sumup/circuit-ui';
 
 const HEADERS = ['Breakpoint name', 'Query'];
 
-const renderBreakpoint = (bp) => {
-  if (typeof bp === 'number') {
-    return `(min-width: ${bp.toString()}px)`;
-  }
```
Comment (connor-baer) [inline .storybook/components/MediaQueriesTable.tsx:25]: There are no more numeric breakpoints since #1039.

File: .storybook/preview-head.html [hunk 2]
```diff
@@ -1,7 +1,388 @@
...
+    --cui-border-promo: #9e33e0;
+    --cui-border-promo-hovered: #8a1ecc;
+    --cui-border-promo-pressed: #7219a9;
+    --cui-border-promo-disabled: rgba(149, 53, 208, 0.4);
+    /* Special colors */
+    --cui-bg-overlay: rgba(0, 0, 0, 0.4);
+    --cui-border-focus: #ebf4ff;
+  }
+
+  @media (prefers-color-mode: dark) {
```
Comment (connor-baer) [inline .storybook/preview-head.html:181]: This is a pseudo dark theme for the time being. It only switches the normal foreground and background colors. This should be updated with the real dark theme once we have it.

File: docs/contributing/2-code-conventions.mdx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { Meta } from '../../.storybook/components';
 
-<Meta title="Introduction/Contributing/Code Conventions" />
```
Comment (connor-baer) [inline docs/contributing/2-code-conventions.mdx:3]: "Contributing" is now its own section.

File: .changeset/strange-bats-listen.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/icons': minor
+---
+
+Added the `IconsManifest` type to the exports. It describes the shape of `@sumup/icons/manifest.json`.
```
Comment (robinmetral) [inline .changeset/strange-bats-listen.md:5]: Surprising to find here but a nice idea 👌😛

File: .storybook/components/MediaQueriesTable.tsx (hunk 1 above)
Comment (robinmetral) [reply .storybook/components/MediaQueriesTable.tsx:25]: 🧹

File: .storybook/main.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    name: '@storybook/react-webpack5',
+    options: {},
+  },
+  babel: async (options) => ({
+    ...options,
+    presets: [
+      ...options.presets,
+      // HACK: Storybook includes `@babel/preset-react` by default, which
+      // overrides the custom preset configuration in `babel.config.json`.
+      // This override overrides the override.
```
Comment (robinmetral) [inline .storybook/main.ts:51]: 😂

File: docs/introduction/1-welcome.mdx [hunk 6]
```diff
@@ -23,35 +23,35 @@ import { Meta, Image, Intro, Teaser } from '../../.storybook/components';
 
 Set up a new app with Circuit UI or add it to an existing project.
 
-[Get started](Introduction/Getting-Started/Page)
+[Get started →](Introduction/Getting-Started/Docs)
```
Comment (robinmetral) [inline docs/introduction/1-welcome.mdx:26]: "get started rightwards arrow, link"

(this is fine here, in prod I would have suggested putting it in ::after 😛)

File: docs/introduction/1-welcome.mdx [hunk 7]
```diff
@@ -61,4 +61,4 @@ If you feel another member of the community violated our CoC or you are experien
 
 ## About SumUp
 
-[SumUp](https://sumup.com) is a mobile-point of sale provider. It is our mission to make easy and fast card payments a reality across the _entire_ world. You can pay with SumUp in more than 30 countries, already. Our engineers work in Berlin, Cologne, Sofia, and Sāo Paulo. We write code in JavaScript, Swift, Ruby, Elixir, Erlang, and much more. Want to come and work with us? [Head to our careers page](https://sumup.com/careers) to find out more.
+[SumUp](https://sumup.com) is a mobile-point of sale provider. It is our mission to make easy and fast card payments a reality across the _entire_ world. You can pay with SumUp in more than 30 countries, already. Our engineers work in Berlin, Copenhagen, Kiev, Sofia, and Sāo Paulo. We write code in TypeScript, Swift, Go, Elixir, Erlang, and much more. Want to come and work with us? [Head to our careers page](https://sumup.com/careers) to find out more.
```
Comment (robinmetral) [inline docs/introduction/1-welcome.mdx:64]: 🆕 🤩 

Maybe we can also update this in the OSS template if this is stiff a thing?

**Review by robinmetral — APPROVED** [review]
This is great! I like the new syntax and agree that we can look into using CSF 3 in a follow-up PR.

And lots of new TS in Storybook components and config! 🎁 I'll admit that I looked at these very briefly, the preview looks good (I just added a[ small comment](https://oss-circuit-ui-git-chore-storybook-7.sumup-vercel.app/?path=/docs/typography-anchor--docs#thread-id=6E-HJ) to the Vercel preview), I think we can move ahead and fix details (the obligatory broken links etc) in the next few days

File: .storybook/main.ts (hunk 5 above)
Comment (robinmetral) [reply .storybook/main.ts:51]: Maybe I misunderstood the migration guide (only skimmed through), but could we use the new [`jsxOptions`](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#dropped-addon-docs-manual-babel-configuration) instead? (but I do love this comment)

File: .changeset/strange-bats-listen.md (hunk 4 above)
Comment (connor-baer) [reply .changeset/strange-bats-listen.md:5]: I needed it for the Storybook components 🤷‍♂️

File: .storybook/main.ts (hunk 5 above)
Comment (connor-baer) [reply .storybook/main.ts:51]: I was very proud when I wrote this 😄

File: docs/introduction/1-welcome.mdx (hunk 7 above)
Comment (connor-baer) [reply docs/introduction/1-welcome.mdx:64]: Honestly, I think this should be rewritten completely. I just wanted to fix the most obvious errors 🤷‍♂️

File: .storybook/main.ts (hunk 5 above)
Comment (connor-baer) [reply .storybook/main.ts:51]: I completely missed that option; thanks for pointing it out! I'll give it a shot.

Comment (connor-baer) [reply .storybook/main.ts:51]: I tried adding...

```ts
  jsxOptions: {
    runtime: 'automatic',
    importSource: '@emotion/react',
  },
```

...in `main.ts`, with no success. The migration guide mentions this under the `addon-docs`, so wonder if this is specific to docs mode.

File: docs/introduction/1-welcome.mdx (hunk 6 above)
Comment (connor-baer) [reply docs/introduction/1-welcome.mdx:26]: Excellent point. Addressed in 63029059.

File: .storybook/main.ts (hunk 5 above)
Comment (robinmetral) [reply .storybook/main.ts:51]: Yeah I was also wondering this. Let's keep the wonderful comment 👌

File: docs/introduction/1-welcome.mdx (hunk 7 above)
Comment (robinmetral) [reply docs/introduction/1-welcome.mdx:64]: Makes sense!

**Pull request merged**